File: dashboard/backend/app/schemas/metrics.py
from pydantic import BaseModel


class MetricDatapoint(BaseModel):
    timestamp: str
    value: float


class MetricTimeseries(BaseModel):
    resource_id: str
    resource_name: str
    metric_name: str
    unit: str
    datapoints: list[MetricDatapoint]


class MetricSummaryItem(BaseModel):
    resource_id: str
    resource_name: str
    metric_name: str
    avg: float
    max: float
    min: float
    p95: float
    unit: str


class IdleResource(BaseModel):
    resource_id: str
    resource_name: str
    service_type: str
    monthly_cost: float
    avg_cpu: float
    avg_network_in: float
    avg_network_out: float
    reason: str

File: dashboard/frontend/src/types/metrics.ts
export interface MetricDatapoint {
  timestamp: string
  value: number
}

export interface MetricTimeseries {
  resource_id: string
  resource_name: string
  metric_name: string
  unit: string
  datapoints: MetricDatapoint[]
}

export interface MetricSummaryItem {
  resource_id: string
  resource_name: string
  metric_name: string
  avg: number
  max: number
  min: number
  p95: number
  unit: string
}

export interface IdleResource {
  resource_id: string
  resource_name: string
  service_type: string
  monthly_cost: number
  avg_cpu: number
  avg_network_in: number
  avg_network_out: number
  reason: string
}
